pkg/syncer/storagepool/migrationController.go

```diff
@@ -70,7 +69,7 @@ func (m *migrationController) relocateCNSVolume(ctx context.Context, volumeID st
 	if !found || err != nil {
 		return fmt.Errorf("failed to find datastoreUrl in StoragePool %s", targetSPName)
 	}
-	dsInfo, err := cnsvsphere.GetDatastoreInfoByURL(ctx, m.vc, m.clusterID, datastoreURL)
+	dsInfo, err := cnsvsphere.GetDatastoreInfoByURL(ctx, m.vc, m.clusterIDs[0], datastoreURL)
```

Comment:
why are we only picking the 1st clusterID?
As pointed out above we should maintain a cache to quickly find the right clusterID for a given datastore.

Reply:
This was just to ignore the errors while I was testing other workflows. I'll look into this and update the code accordingly.